test/runtests.jl

```diff
@@ -48,39 +48,39 @@ let s = "a string チャネルパートナーの選択"
     close(p)
 end
 
-@test_throws ErrorException encode("qwertyé€", "ASCII")
+@test_throws InvalidSequenceError encode("qwertyé€", "ASCII")
 try
     encode("qwertyé€", "ASCII")
 catch err
      io = IOBuffer()
      showerror(io, err)
      @test takebuf_string(io) ==
-        "iconv error: byte sequence 0xc3a9e282ac is invalid in source encoding or cannot be represented in target encoding"
+        "iconv: byte sequence 0xc3a9e282ac is invalid in source encoding or cannot be represented in target encoding"
```

Comment:
Actually, I'm not even sure we should print the iconv function name in most cases. For errors like "invalid sequence", which are well-understood and can happen with a pure-Julia implementation, I would simply print the interesting part. Only for unexpected/generic errors, I would print the implementation details (like function and error code) because we don't know what happened.